x/ibc/core/02-client/types/genesis.go

```diff
@@ -149,12 +171,23 @@ func (gs GenesisState) Validate() error {
 			if err := cs.ValidateBasic(); err != nil {
 				return fmt.Errorf("invalid client consensus state %v index %d: %w", cs, i, err)
 			}
+
+			// ensure consensus state type matches client state type
+			if clientState.ClientType() != cs.ClientType() {
+				return fmt.Errorf("consensus state client type %s does not equal client state client type %s", cs.ClientType(), clientState.ClientType())
+			}
+
 		}
+
 	}
 
 	if gs.CreateLocalhost && !gs.Params.IsAllowedClient(exported.Localhost) {
 		return fmt.Errorf("localhost client is not registered on the allowlist")
 	}
 
+	if maxSequence != 0 && maxSequence >= gs.NextClientSequence {
```

Comment:
Should we be checking that if `maxSequence == 0` then `gs.NextClientSequence == 0` as well?

And should we check that if `maxSequence != 0`, then `gs.NextClientSequence == maxSequence + 1`?

Or no?

Reply:
I don't think we need to put such strict restrictions on this. I think there is value in allowing the next client sequence to be very different from the maximum sequence. For example, if during a genesis migration, validators decide to prune old expired clients. Or if genesis validators decide to create clients using identifiers in genesis from 0-900 and start regular clients at 1000 to distinguish their origins